dietpi/dietpi-software

```diff
@@ -8385,6 +8414,118 @@ _EOF_
 			mkdir -p /etc/exports.d
 			[[ -f '/etc/exports.d/dietpi.exports' ]] || echo '/mnt/dietpi_userdata *(rw,async,no_root_squash,fsid=0,crossmnt,no_subtree_check)' > /etc/exports.d/dietpi.exports
 
+		fi 
+    
+		software_id=182 # Unbound
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+		
+		Banner_Configuration
+
+		# Download list of root hints
+		Download_Install https://www.internic.net/domain/named.root /var/lib/unbound/root.hints
+
+		cat << _EOF_ > /etc/unbound/unbound.conf.d/dietpi.conf
+server:
+    verbosity: 0
+    
+    interface: 0.0.0.0
+    port: 53
+    do-ip4: yes
+    do-udp: yes
+    do-tcp: yes
+    
+    do-ip6: yes
+
+    prefer-ip6: no
+
+    root-hints: "/var/lib/unbound/root.hints"
+
+    harden-glue: yes
+    harden-large-queries: yes
+
+    harden-dnssec-stripped: yes
+
+    use-caps-for-id: yes
+
+    edns-buffer-size: 1472
+
+    rrset-roundrobin: yes
+
+    cache-min-ttl: 300
+    cache-max-ttl: 86400
+
+    serve-expired: yes
+
+    harden-algo-downgrade: yes
+
+    harden-short-bufsize: yes
+
+    hide-identity: yes
+
+    identity: "Server"
+
+    hide-version: yes
+
+    do-daemonize: no
+
+    neg-cache-size: 4M
+
+    qname-minimisation: yes
+
+    minimal-responses: yes
+
+    prefetch: yes
+    prefetch-key: yes
+
+    num-threads: 1
+
+    msg-cache-size: 50m
+    rrset-cache-size: 100m
+
+    so-reuseport: yes
+
+    so-rcvbuf: 4m
+    so-sndbuf: 4m
+
+    unwanted-reply-threshold: 10000
+
+    ratelimit: 1000
+
+    log-queries: no
+    log-replies: no
+    logfile: ''
+    
+    access-control: 0.0.0.0/0 refuse
+    access-control: 127.0.0.1 allow
+    access-control: $(hostname -I | grep -o '192.168.[[:digit:]]').0/24 allow
+
+    private-address: 192.168.0.0/16
+    private-address: 169.254.0.0/16
+    private-address: 172.16.0.0/12
+    private-address: 10.0.0.0/8
+    private-address: fd00::/8
+    private-address: fe80::/10
+_EOF_
+    
+		# Configuration
+		if grep -q 'CONFIG_ENABLE_IPV6=0' /boot/dietpi.txt
+		then
+			G_CONFIG_INJECT 'do-ip6:[[:blank:]]' '    do-ip6: no' /etc/unbound/unbound.conf.d/dietpi.conf
+		fi
+		if grep -q 'CONFIG_PREFER_IPV4=0' /boot/dietpi.txt
+		then
+			G_CONFIG_INJECT 'prefer-ip6:[[:blank:]]' '    prefer-ip6: yes' /etc/unbound/unbound.conf.d/dietpi.conf
+		fi
+		
+		if (( ${aSOFTWARE_INSTALL_STATE[93]} > 0 )); then
+		G_DIETPI-NOTIFY 2 'Configuring Unbound to work with Pi-hole'
+			cat << '_EOF_' > /etc/unbound/unbound.conf.d/dietpi-pihole.conf
+port: 5353
+interface: 127.0.0.1
+_EOF_
+			G_CONFIG_INJECT 'PIHOLE_DNS_1=' 'PIHOLE_DNS_1=127.0.0.1#5353' /etc/pihole/setupVars.conf
+			G_CONFIG_INJECT 'PIHOLE_DNS_2=' 'PIHOLE_DNS_2=' /etc/pihole/setupVars.conf
+		fi
```

Comment:
Let's restart unbound once after configuring it. That way it can be used without reboot, e.g. when doing it via CLI, the reboot is not forced:
```
G_EXEC systemctl restart unbount
```